### ISSUE TITLE
[Feature Request] Use Angular 2 cli project with ExpressJs

### ISSUE THREAD

**Issue by hamzatrq** [angular/angular-cli#2173]
I have seen a few seed projects with angular 2 and expressjs. But what I want is the project created with angular cli to be compatible with expressjs generated through express generator. So that I can use the features of angular cli. I posted a question on Stack Overflow 3 months back but still didn't got any answer. That's why I am posting it as an issue here.

**Comment by filipesilva** [comment]
Integrating a custom server isn't something we're looking at doing now. We do have addon functionality planned, so an addon could perhaps do that.

**Comment by kylecordes** [comment]
Incidentally this exact same question came up in class I was teaching today. I suggested three possible relatively easy solutions:

1. There is a proxy feature suitable if you're just trying to integrate for the purpose of reaching an API. I believe it was added by the PR linked here 
https://github.com/angular/angular-cli/pull/1487/files  , which unfortunately doesn't include any documentation, but it should be easy enough to figure out.
2. Simply make your existing express based server proxy back to CLI ```ng serve``` while in development mode. This takes just a handful of lines of code, there is an NPM module which easily adds proxying to an express based project.
3. Use ng watch rather than ng serve, and have your existing express server serve the files you're keeping up-to-date using CLI.

Are any or all of these worthy of adding to the documentation, either in the readme or in a FAQ or something? I would be willing to do so with a nudge toward which of these is most preferred by the CLI team.
